### PR TITLE
Student detail view and statistics

### PR DESCRIPTION
**Pull request by slaviana88**

### CODE REVIEW

File: clock.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import subprocess
 from apscheduler.schedulers.blocking import BlockingScheduler
 
-sched = BlockingScheduler()
 
+sched = BlockingScheduler()
```
Comment (RadoRado) [inline clock.py]: Checkout `clock.py`

File: loki/education/views/teacher_dashboard.py [hunk 2]
```diff
@@ -29,6 +30,40 @@ def get_context_data(self, **kwargs):
...
+    pk_url_kwarg = 'student'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        course = Course.objects.get(id=self.kwargs.get("course"))
+        tasks = Task.objects.filter(course=course, gradable=True)
+        url_tasks = Task.objects.filter(course=course, gradable=False)
+
+        student = Student.objects.get(id=self.object.user.id)
+        context['passed_solutions'] = Solution.objects.filter(student=student, task__in=tasks, status=2).count()
```
Comment (RadoRado) [inline loki/education/views/teacher_dashboard.py]: `status=2` should be something more readable. Use `Solution.SOME_STATUS`

File: loki/education/views/teacher_dashboard.py [hunk 3]
```diff
@@ -29,6 +30,40 @@ def get_context_data(self, **kwargs):
...
+        context['failed_solutions'] = Solution.objects.filter(student=student, task__in=tasks, status=3).count()
```
Comment (RadoRado) [inline loki/education/views/teacher_dashboard.py]: `status=3` should be something more readable. Use `Solution.SOME_STATUS`

File: loki/education/views/teacher_dashboard.py [hunk 4]
```diff
@@ -29,6 +30,40 @@ def get_context_data(self, **kwargs):
...
+        context['url_solutions'] = Solution.objects.filter(student=student, task__in=url_tasks).count()
+        context['count_solutions'] = context['passed_solutions'] + context['failed_solutions'] + context['url_solutions']
+
+        course_presence = {}
```
Comment (RadoRado) [inline loki/education/views/teacher_dashboard.py]: Can we move this to something like a service? I think it repeats itself with the other views. Something like:

```python
context['course_presence'] = get_course_presence(course=course, student=student)
```

File: loki/static/js/main.js [hunk 5]
```diff
@@ -112,4 +112,9 @@ $(document).ready(function(){
   $(".student-note").click(function(){
     $(this).closest('.student-box').find('.student-notes-table').toggle();
   });
+
+  $(".student-presence-percent").each(function(){
+    var percent = parseInt($(this).html())/100;
+    $(this).css("color", "rgb(" + Math.floor(255 * (1-percent)) + "," + Math.floor(255 * percent) + ",0)");
```
Comment (RadoRado) [inline loki/static/js/main.js]: Add comment here. What is this code doing?

File: loki/education/views/teacher_dashboard.py [hunk 6]
```diff
@@ -29,6 +30,40 @@ def get_context_data(self, **kwargs):
...
+                        IsTeacherMixin,
+                        CannotSeeOthersCoursesDashboardsMixin,
+                        DetailView):
+
+    model = CourseAssignment
+    pk_url_kwarg = 'student'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        course = Course.objects.get(id=self.kwargs.get("course"))
```
Comment (RadoRado) [inline loki/education/views/teacher_dashboard.py]: `get_object_or_404` here.

File: loki/static/js/main.js (hunk 5 above)
Comment (slaviana88) [inline loki/static/js/main.js]: hmmm I will ask Kamen for it

File: loki/education/views/teacher_dashboard.py (hunk 6 above)
Comment (slaviana88) [inline loki/education/views/teacher_dashboard.py]: look at mixins.

File: clock.py (hunk 1 above)
Comment (slaviana88) [inline clock.py]: fixed

File: loki/education/views/teacher_dashboard.py (hunk 3 above)
Comment (slaviana88) [inline loki/education/views/teacher_dashboard.py]: fixed

File: loki/education/views/teacher_dashboard.py (hunk 2 above)
Comment (slaviana88) [inline loki/education/views/teacher_dashboard.py]: fixed

**Pull request merged**